### DISCUSSION TITLE
Дата обновления в описании плейлиста

### DISCUSSION THREAD

**Discussion by Sairon92** [Chimildic/goofy#13] [Помощь с ошибкой] [answered]
Добрый день. 
Что бы получить треки, которые слушал, за последние 14 дней пишу код:
```js
function createHistoryPlaylist2() {
    let RsntHistoryTracks = RecentTracks.get();
    Filter.rangeDateRel(RsntHistoryTracks, 14);
    console.log('Количество треков RH', RsntHistoryTracks.length);
    let UpdTime = new Date();
    Playlist.saveWithReplace({
        id: 'abc',
        name: 'History 14 by Goofy',
        description: UpdTime,
        tracks: RsntHistoryTracks,
    });
}

```
Если убрать в создании плейлиста `description: UpdTime`, то все работает хорошо, в противном случае выдает ошибку:

> TypeError: array.slice is not a function (строка 977, файл Library)

**Comment by Chimildic** [comment, marked as answer]
Перед отправкой, описание обрезается до 300 символов. Ошибка возникает из-за того, что передается не строковое значение, а дата. Чтобы перевести в строку, используйте _один_ из вариантов:
```js
new Date().toLocaleDateString(); // 23.12.2020
new Date().toLocaleString(); // 23.12.2020, 11:33:49
```
Если нужен другой формат даты, уточните. Посмотрим что можно сделать.

**Reply by Sairon92** [reply]
Спасибо. Помогло

**Reply by Bedmit** [reply]
А как записать, чтобы был и текст, и только время. 
Например - "Обновлено в 15:33" .  
Пока получается либо то, либо другое. Причем время только в 12-ти часовом формате.

**Reply by Chimildic** [reply]
> Например - "Обновлено в 15:33" 
```js
let text = 'Обновлено в ' + new Date().toTimeString().substring(0, 5);
```

**Reply by Bedmit** [reply]
Отлично! Теперь, то что надо. Спасибо.
ИМХО, для таких неучей как я, только примеры понятны. Чтобы просто копировать и вставлять к себе в скрипт.

**Reply by Chimildic** [reply]
> только примеры понятны. Чтобы просто копировать и вставлять к себе в скрипт.

Слишком много возможностей, чтобы все обложить примерами. Пусть собирается хотя бы постепенно.

**Reply by sergvpr** [reply]
`Например - "Обновлено в 15:33"`
Как изменить часовой пояс или как можно добавить 3 часа ко времени в этом случае?
Спасибо.

**Reply by Chimildic** [reply]
@sergvpr ответил ниже

**Comment by Chimildic** [comment]
По поводу таймзоны. Можно использовать следующий код.
```js
new Date().toLocaleString("ru-RU", { timeZone: "Asia/Omsk" }).match(/\d+:\d+/i)[0];
```

Возможные зоны ниже. Ориентируйтесь на смещение относительно Москвы `MSK`.
```
Europe/Kaliningrad	MSK-01 - Kaliningrad
Europe/Moscow	MSK+00 - Moscow area
Europe/Kirov	MSK+00 - Kirov
Europe/Volgograd	MSK+00 - Volgograd
Europe/Astrakhan	MSK+01 - Astrakhan
Europe/Saratov	MSK+01 - Saratov
Europe/Ulyanovsk	MSK+01 - Ulyanovsk
Europe/Samara	MSK+01 - Samara, Udmurtia
Asia/Yekaterinburg	MSK+02 - Urals
Asia/Omsk	MSK+03 - Omsk
Asia/Novosibirsk	MSK+04 - Novosibirsk
Asia/Barnaul	MSK+04 - Altai
Asia/Tomsk	MSK+04 - Tomsk
Asia/Novokuznetsk	MSK+04 - Kemerovo
Asia/Krasnoyarsk	MSK+04 - Krasnoyarsk area
Asia/Irkutsk	MSK+05 - Irkutsk, Buryatia
Asia/Chita	MSK+06 - Zabaykalsky
Asia/Yakutsk	MSK+06 - Lena River
Asia/Khandyga	MSK+06 - Tomponsky, Ust-Maysky
Asia/Vladivostok	MSK+07 - Amur River
Asia/Ust-Nera	MSK+07 - Oymyakonsky
Asia/Magadan	MSK+08 - Magadan
Asia/Sakhalin	MSK+08 - Sakhalin Island
Asia/Srednekolymsk	MSK+08 - Sakha (E); North Kuril Is
Asia/Kamchatka	MSK+09 - Kamchatka
Asia/Anadyr	MSK+09 - Bering Sea
```

**Reply by amd64fox** [reply]
А как сделать такой вариант 11.01.2021, 0:55
Вариант выше  `new Date().toLocaleString();` выдает такое 1/11/2021, 12:55:01 AM
в дате идет первый месяц и время в 12 часовом формате

**Reply by Chimildic** [reply]
Странно,  у меня без дополнительных настроек определяет все правильно. Может из-за сервера, который мне подбирает гугл. Попробуйте так:
```js
new Date().toLocaleString("ru-RU", { timeZone: "Asia/Omsk", hour12: false }).slice(0, -3)
```

**Reply by amd64fox** [reply]
Да спасибо, так заработало.
